### PR TITLE
fix(TDI-46340): tMongoDB convert Bson to Json.

### PR DESCRIPTION
**Pull request by pyzhou-talend**
**Please check if the PR fulfills these requirements**

- [ ] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format) ?
- [ ] Unit tests for the Java changes have been added (for bug fixes / features) ?
- [ ] TUJ for the JavaJet changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The new code does not introduce new technical issues

**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-46340

**What is the new behavior?**

**BREAKING CHANGE**

If this PR contains a breaking change, please describe the impact and migration
path for existing applications.
If not please remove this section.

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.designer.components.bigdata/components/tMongoDBInput/tMongoDBInput_begin.javajet [hunk 1]
```diff
@@ -266,6 +267,23 @@
 								value=currentObj.get(currentName);
 							}
 						}
+						<%if( (version_3_5_X || version_4_4_X ) && bsonToString ){%>
```
Comment (vyu-talend) [inline main/plugins/org.talend.designer.components.bigdata/components/tMongoDBInput/tMongoDBInput_begin.javajet:270]: Hello, Pengyu. What if the value neither instance of org.bson.Document nor List?

Comment (pyzhou-talend) [reply main/plugins/org.talend.designer.components.bigdata/components/tMongoDBInput/tMongoDBInput_begin.javajet:270]: Not sure I get what you mean. if the value neither instance of org.bson.Document nor List, the value should be a basic value like Int, double then, no convertion need be done.

Comment (vyu-talend) [reply main/plugins/org.talend.designer.components.bigdata/components/tMongoDBInput/tMongoDBInput_begin.javajet:270]: Sounds reasonable.

**Review by vyu-talend — APPROVED** [review]

**Review by jzhao-talend — DISMISSED** [review]
LGTM

**Review by jzhao-talend — APPROVED** [review]
LGTM

**Pull request merged**